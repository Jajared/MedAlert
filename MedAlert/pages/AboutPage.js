import { SafeAreaView, View, Text, StatusBar, StyleSheet } from "react-native";
import BackNavBar from "../components/BackNavBar/BackNavBar";

export default function AboutPage({ navigation }) {
  return (
    <SafeAreaView style={styles.container}>
      <StatusBar barStyle="dark-content" />
      <BackNavBar navigation={navigation} title="About" />
      <Text>We are a group of students from NUS</Text>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "column",
    alignItems: "center",
    backgroundColor: "#fff",
    flex: 1,
    width: "100%",
  },
});
