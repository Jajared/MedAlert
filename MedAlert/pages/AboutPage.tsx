import { SafeAreaView, View, Text, StatusBar, StyleSheet, Image } from "react-native";
import BackNavBar from "../components/BackNavBar";

export default function AboutPage({ navigation }) {
  return (
    <SafeAreaView style={styles.container}>
      <StatusBar barStyle="dark-content" />
      <BackNavBar navigation={navigation} title="About" />
      <View style={styles.headerSection}>
      <Text style={{fontSize: 16, color: "#5A5A5A"}}>{`Forgetting to take medication is a common problem, especially in a country like Singapore, which has so many people diagnosed with chronic illnesses (such as diabetes and hypertension) who rely on medication every single day. Various medications must be consumed at various frequencies which makes consumption tracking tedious and confusing.\n\nMedAlert aims to provide a user-friendly solution for individuals to manage their medication schedules efficiently.`} 
</Text>
      </View>
<View style={styles.referenceSection}>
  <Text>References:</Text>
  <View style={styles.referenceIcons}>
  <Image source={require("../assets/data-gov.png")} style={styles.icon}/>
  <Image source={require("../assets/apimedic-icon.png")} style={styles.icon}/>
  </View>
  
</View>
<View style={{flex:1}}/>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: "column",
    alignItems: "center",
    backgroundColor: "#fff",
    flex: 1,
    width: "100%",
  }, headerSection: {
    width: "90%",
    flex: 2
  },
    referenceSection: {
      flex:2,
    width: "90%",
  }, referenceIcons: {
    flexDirection:"row"
  },icon: {
    flex:1,
    margin: 10,
    height: 120,
    resizeMode: "contain",
  }
});
