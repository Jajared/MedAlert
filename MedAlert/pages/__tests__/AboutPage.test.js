import { render, screen, fireEvent } from "@testing-library/react-native";
import AboutPage from "../AboutPage";
import { expect, jest, test } from "@jest/globals";

jest.mock("expo-font");
jest.mock("expo-asset");

const navigationObject = {
  navigate: jest.fn(),
  goBack: jest.fn(),
};

describe("AboutPage", () => {
  // Check if renders correctly
  it("renders correctly", () => {
    const tree = render(<AboutPage navigation={navigationObject} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  // Check number of children
  it("has 5 children", () => {
    render(<AboutPage navigation={navigationObject} />);
    expect(screen.getByTestId("AboutPage").children.length).toBe(5);
  });
});
