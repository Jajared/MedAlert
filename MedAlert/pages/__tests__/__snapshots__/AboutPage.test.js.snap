// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutPage/> renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "paddingHorizontal": 16,
        "width": "90%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginLeft": -20,
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "black",
              "fontSize": 24,
            },
            undefined,
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        ï„¤
      </Text>
    </View>
    <Text
      style={
        {
          "fontSize": 20,
          "fontWeight": "bold",
          "marginLeft": 16,
        }
      }
    >
      About
    </Text>
  </View>
  <Text>
    We are a group of students from NUS
  </Text>
</RCTSafeAreaView>
`;
